### PR TITLE
Allow campaign data download

### PR DESCRIPTION
**Pull request by jacopofar**
This version downloads two files per day.

One is the __keyword report__, that previously was downloaded with the name __ad_performance.csv.gz__, except for the name structure of the file is the same.

The other one is the __ad performance report__ which contains now the performance report from Bing ads including, among other things, the labels.

The CLI arguments and the README have been changed accordingly and two file names can be specified.

### CODE REVIEW

File: bingads_downloader/config.py [hunk 1]
```diff
@@ -47,11 +55,11 @@ def timeout() -> int:
...
-    """The attempts to download a single file in case of HTTP errors or timeouts"""
+def total_attempts_for_single_day() -> int:
+    """The attempts to download a single day (ad and keyword performance) in case of HTTP errors or timeouts"""
     return 5
 
 
 def retry_timeout_interval() -> int:
     """number of seconds to wait before trying again to download a single day"""
-    return 10
+    return 10
```
Comment (Tafkas) [inline bingads_downloader/config.py]: not PEP8 😉

Comment (jacopofar) [reply bingads_downloader/config.py]: yep, fixed

File: bingads_downloader/config.py [hunk 2]
```diff
@@ -8,11 +8,19 @@ def data_dir() -> str:
...
     return 'ad_performance.csv.gz'
 
 
+def keyword_performance_data_file() -> str:
+    """The name of the file the keyword performance result is written to"""
+    return 'keyword_performance.csv.gz'
+
+def campaign_performance_data_file() -> str:
+    """The name of the file the campaign performance result is written to"""
+    return 'campaign_performance.csv.gz'
```
Comment (alexandschaefer) [inline bingads_downloader/config.py]: Campaign performance is not mentioned in the readme or CLI. Is there a reason?

Comment (jacopofar) [reply bingads_downloader/config.py]: No, I forgot to remove it after removing that logic. Removed now

File: bingads_downloader/downloader.py [hunk 3]
```diff
@@ -47,39 +47,56 @@ def download_data_sets(api_client: BingReportClient):
...
                           .format(date=current_date, elapsed=time.time() - start_time))
+                    start_time = time.time()
+                    print('About to download keyword data for {date:%Y-%m-%d}'
+                          .format(date=current_date))
+                    submit_and_download(report_request_keyword, api_client, str(filepath), config.keyword_performance_data_file())
+                    print('Successfully downloaded keyword data for {date:%Y-%m-%d} in {elapsed:.1f} seconds'
+                      .format(date=current_date, elapsed=time.time() - start_time))
+                    print('About to download campaign data for {date:%Y-%m-%d}'
+                          .format(date=current_date))
+                    submit_and_download(report_request_campaign, api_client, str(filepath), config.campaign_performance_data_file())
```
Comment (alexandschaefer) [inline bingads_downloader/downloader.py]: module 'bingads_downloader.config' has no attribute 'campaign_performance_data_file'

Comment (jacopofar) [reply bingads_downloader/downloader.py]: Put back, now with the documentation and the CLI command

File: README.md [hunk 4]
```diff
@@ -1,16 +1,16 @@
 # BingAds Performance Downloader
 
-A Python script for downloading performance data from the [BingAds API version 11](https://msdn.microsoft.com/en-us/library/bing-ads-overview(v=msads.100).aspx) to local files. The code is largely based on [Bing Ads Python SDK](https://github.com/BingAds/BingAds-Python-SDK).
+A Python script for downloading performance and keyword data from the [BingAds API version 11](https://msdn.microsoft.com/en-us/library/bing-ads-overview(v=msads.100).aspx) to local files. The code is largely based on [Bing Ads Python SDK](https://github.com/BingAds/BingAds-Python-SDK).
```
Comment (alexandschaefer) [inline README.md]: this is the current link https://docs.microsoft.com/en-us/bingads/guides/?view=bingads-11

Comment (Tafkas) [reply README.md]: Looks like it

**Pull request merged**